bittensor/_receptor/receptor_pool_impl.py

```diff
@@ -133,28 +191,100 @@ def forward (
         # Init function.
         def call_forward( args ):
             return args['receptor'].forward( args['synapses'], args['inputs'], args['timeout'] )
+        
+        # Unpack responses
+        results_dict = dict(
+            outputs=[],
+            codes=[],
+            times=[],
+            uids=[]
+        )
+
+
+        # resolve the minumum number of successes
+        if min_success == None:
+            min_success = len(endpoints)
+        if min_success < 1:
+            min_success = int(min_success*len(endpoints))
+        elif min_success >= 1:
+            min_success = int(min(min_success,len(endpoints)))
+        elif min_success <= 0:
+            raise Exception(' REQUIRED: 0<min_success<len(endpoints)')
+
+
+        # ensure max_workers
+        if not isinstance(max_workers, int):
+            max_workers = len(endpoints)
+
 
         # Submit calls to receptors.
-        with concurrent.futures.ThreadPoolExecutor( max_workers = len(endpoints) ) as executor:
-            responses = executor.map( call_forward, call_args, timeout=10*timeout)
-        
-        # Release semephore.
-        for receptor in receptors:
-            receptor.semaphore.release()
+        with concurrent.futures.ThreadPoolExecutor( max_workers = max_workers ) as executor:
```

Comment:
This is actually a bug with the previous receptor pool implementation. Rather than creating a separate threadpool with each forward pass, it should be using the threadpool within the receptor pool already. Do you want to include this bug fix as well? :)